File: mobile/lib/models/frame_model.dart
class FrameModel {
  final int id;
  final String name;
  final String layoutType;
  final String? thumbnailPath;
  final bool isActive;

  FrameModel({
    required this.id,
    required this.name,
    required this.layoutType,
    this.thumbnailPath,
    this.isActive = true,
  });

  factory FrameModel.fromJson(Map<String, dynamic> json) {
    return FrameModel(
      id: json['id'],
      name: json['name'],
      layoutType: json['layout_type'] ?? '4-cut',
      thumbnailPath: json['thumbnail_path'],
      isActive: (json['is_active'] ?? 1) == 1,
    );
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'layout_type': layoutType,
        'thumbnail_path': thumbnailPath,
      };
}

File: backend/src/models/frameModel.js
const db = require('../config/database');

const FrameModel = {
  findAll() {
    return db.prepare(`SELECT * FROM frames WHERE is_active = 1 ORDER BY id DESC`).all();
  },

  findById(id) {
    return db.prepare(`SELECT * FROM frames WHERE id = ?`).get(id);
  },

  create({ name, layout_type, thumbnail_path }) {
    const info = db
      .prepare(`INSERT INTO frames (name, layout_type, thumbnail_path) VALUES (?, ?, ?)`)
      .run(name, layout_type || '4-cut', thumbnail_path || null);
    return this.findById(info.lastInsertRowid);
  },

  update(id, { name, layout_type, thumbnail_path }) {
    db.prepare(
      `UPDATE frames SET name = COALESCE(?, name), layout_type = COALESCE(?, layout_type), thumbnail_path = COALESCE(?, thumbnail_path) WHERE id = ?`
    ).run(name, layout_type, thumbnail_path, id);
    return this.findById(id);
  },

  softDelete(id) {
    db.prepare(`UPDATE frames SET is_active = 0 WHERE id = ?`).run(id);
    return true;
  },
};

module.exports = FrameModel;
